src/main.rs

```diff
@@ -157,33 +157,55 @@ async fn main() -> anyhow::Result<()> {
-            .await
-            {
-                Ok(fd) => {
-                    // Update the config here
-                    config = config
-                        .with_l1_gas_price(Some(fd.l1_gas_price))
-                        .with_l2_gas_price(Some(fd.l2_fair_gas_price))
-                        .with_l1_pubdata_price(Some(fd.fair_pubdata_price))
-                        .with_price_scale(Some(fd.estimate_gas_price_scale_factor))
-                        .with_gas_limit_scale(Some(fd.estimate_gas_scale_factor))
-                        .with_chain_id(Some(fd.chain_id.as_u64() as u32));
-                    Some(fd)
+            if let Some(tx_hash) = fork.fork_transaction_hash {
+                // If fork_transaction_hash is provided, use from_network_tx
+                match ForkDetails::from_network_tx(&fork.fork_url, tx_hash, &config.cache_config)
+                    .await
+                {
+                    Ok(fd) => {
+                        config = config
+                            .with_l1_gas_price(Some(fd.l1_gas_price))
+                            .with_l2_gas_price(Some(fd.l2_fair_gas_price))
+                            .with_l1_pubdata_price(Some(fd.fair_pubdata_price))
+                            .with_price_scale(Some(fd.estimate_gas_price_scale_factor))
+                            .with_gas_limit_scale(Some(fd.estimate_gas_scale_factor))
+                            .with_chain_id(Some(fd.chain_id.as_u64() as u32));
+                        Some(fd)
+                    }
+                    Err(error) => {
+                        tracing::error!("cannot fork from transaction: {:?}", error);
+                        return Err(anyhow!(error));
+                    }
                 }
-                Err(error) => {
-                    tracing::error!("cannot fork: {:?}", error);
-                    return Err(anyhow!(error));
+            } else {
+                // Otherwise, use from_network
+                match ForkDetails::from_network(
+                    &fork.fork_url,
+                    fork.fork_block_number,
+                    &config.cache_config,
+                )
+                .await
+                {
+                    Ok(fd) => {
+                        config = config
+                            .with_l1_gas_price(Some(fd.l1_gas_price))
+                            .with_l2_gas_price(Some(fd.l2_fair_gas_price))
+                            .with_l1_pubdata_price(Some(fd.fair_pubdata_price))
+                            .with_price_scale(Some(fd.estimate_gas_price_scale_factor))
+                            .with_gas_limit_scale(Some(fd.estimate_gas_scale_factor))
+                            .with_chain_id(Some(fd.chain_id.as_u64() as u32));
+                        Some(fd)
+                    }
+                    Err(error) => {
+                        tracing::error!("cannot fork: {:?}", error);
+                        return Err(anyhow!(error));
```

Comment:
This part of the flow looks the same for both options, can we construct `ForkDetails` first and then run this after? Just so we can avoid copy-pasting code

Reply:
Updated. I elected to include update config methods to TestNodeConfig impl so that it may be reused.